cmd/osm-controller/certificates.go

```diff
@@ -43,24 +43,9 @@ const (
 	rootCertOrganization = "Open Service Mesh"
 )
 
-// Functions we can call to create a Certificate Manager for each kind of supported certificate issuer
-var certManagers = map[certificateManagerKind]func(kubeClient kubernetes.Interface, kubeConfig *rest.Config, enableDebugServer bool) (certificate.Manager, debugger.CertificateManagerDebugger, error){
-	tresorKind:      getTresorOSMCertificateManager,
-	keyVaultKind:    getAzureKeyVaultOSMCertificateManager,
-	vaultKind:       getHashiVaultOSMCertificateManager,
-	certmanagerKind: getCertManagerOSMCertificateManager,
-}
-
-// Get a list of the supported certificate issuers
-func getPossibleCertManagers() []string {
-	var possible []string
-	for kind := range certManagers {
-		possible = append(possible, string(kind))
-	}
-	return possible
-}
+var validCertificateManagerOptions = []string{tresorKind, vaultKind, certmanagerKind}
```

Comment:
I liked the functional approach of registering the certificate managers factories in this map.
Not opposed to using the `switch case` if that makes it easier for folks to read.

Reply:
I thought that was pretty great too. I decided to opt for this route though when I realized the function signatures don't need to match to generate each kind of manager. I thought this route would also be a little easier for me to add tests for.